src/python/pants/option/arg_splitter_test.py

```diff
@@ -98,140 +86,259 @@ def assert_not_positional(arg):
-                {
-                  '': ['-f'],
-                  'compile.java': ['--long-flag', '-x'],
-                  'compile': ['-g'],
-                  'test.junit': ['-i']
-                },
-                ['src/java/org/pantsbuild/foo', 'src/java/org/pantsbuild/bar:baz'])
-    self._split('./pants -farg --fff=arg compile --gg-gg=arg-arg -g test.junit --iii '
-                '--compile-java-long-flag src/java/org/pantsbuild/foo src/java/org/pantsbuild/bar:baz',
-                ['compile', 'test'],
-                {
-                  '': ['-farg', '--fff=arg'],
-                  'compile': ['--gg-gg=arg-arg', '-g'],
-                  'test.junit': ['--iii'],
-                  'compile.java': ['--long-flag'],
-                },
-                ['src/java/org/pantsbuild/foo', 'src/java/org/pantsbuild/bar:baz'])
-
-  def test_distinguish_goals_from_target_specs(self):
-    self._split('./pants compile test foo::', ['compile', 'test'],
-                {'': [], 'compile': [], 'test': []}, ['foo::'])
-    self._split('./pants compile test foo::', ['compile', 'test'],
-                {'': [], 'compile': [], 'test': []}, ['foo::'])
-    self._split('./pants compile test:test', ['compile'], {'': [], 'compile': []}, ['test:test'])
-    self._split('./pants test test:test', ['test'], {'': [], 'test': []}, ['test:test'])
-    self._split('./pants test ./test', ['test'], {'': [], 'test': []}, ['./test'])
-    self._split('./pants test //test', ['test'], {'': [], 'test': []}, ['//test'])
-
-  def test_descoping_qualified_flags(self):
-    self._split('./pants compile test --compile-java-bar --no-test-junit-baz foo/bar',
-                ['compile', 'test'],
-                {'': [], 'compile': [], 'compile.java': ['--bar'], 'test': [],
-                 'test.junit': ['--no-baz']}, ['foo/bar'])
+    self.assert_valid_split(
+      './pants --compile-java-long-flag -f compile -g compile.java -x test.junit -i '
+      'src/java/org/pantsbuild/foo src/java/org/pantsbuild/bar:baz',
+      expected_goals=['compile', 'test'],
+      expected_scope_to_flags={
+        '': ['-f'],
+        'compile.java': ['--long-flag', '-x'],
+        'compile': ['-g'],
+        'test.junit': ['-i']
+      },
+      expected_positional_args=['src/java/org/pantsbuild/foo', 'src/java/org/pantsbuild/bar:baz'],
+    )
+    self.assert_valid_split(
+      './pants -farg --fff=arg compile --gg-gg=arg-arg -g test.junit --iii '
+      '--compile-java-long-flag src/java/org/pantsbuild/foo src/java/org/pantsbuild/bar:baz',
+      expected_goals=['compile', 'test'],
+      expected_scope_to_flags={
+        '': ['-farg', '--fff=arg'],
+        'compile': ['--gg-gg=arg-arg', '-g'],
+        'test.junit': ['--iii'],
+        'compile.java': ['--long-flag'],
+      },
+      expected_positional_args=['src/java/org/pantsbuild/foo', 'src/java/org/pantsbuild/bar:baz'],
+    )
+
+  def test_distinguish_goals_from_positional_args(self) -> None:
```

Comment:
This function is the same, modulo using keyword args and some formatting tweaks.